docs/migration/fastapi.rst

```diff
@@ -174,8 +174,290 @@ and to easily access dependencies from higher levels.
+            ):
+                # Setup code here
+                yield
+                # Teardown code here
+
+    .. tab-item:: Litestar
+        :sync: litestar
+
+        .. code-block:: python
+
+            @asynccontextmanager
+            async def lifespan(
+                app_settings: AppSettings,
+                _app: Litestar,
+            ):
+                # Setup code here
+                yield
+                # Teardown code here
+
+~~~~~~~
+Cookies
+~~~~~~~
+
+While with FastAPI you usually set cookies on the response `Response` object, in Litestar it will be set at the decorator level.
+
+.. tab-set::
+
+    .. tab-item:: FastAPI
+        :sync: fastapi
+
+        .. code-block:: python
+
+            @app.get("/")
+            async def index(response: Response) -> dict[str, str]:
+                response.set_cookie(key="my_cookie", value="cookie_value")
+                ...
+
+    .. tab-item:: Litestar
+        :sync: litestar
+
+        .. code-block:: python
+
+            @get(response_cookies={"my-cookie": "cookie-value"})
+            async def handler() -> str:
+                ...
+
+
+Dependencies parameters
+~~~~~~~~~~~~~~~~~~~~~~~
+The way dependencies parameters are passed differs between FastAPI and Litestar, note the `state: State` parameter in the Litestar example.
+
+.. tab-set::
+
+    .. tab-item:: FastAPI
+        :sync: fastapi
+
+        .. code-block:: python
+
+            from fastapi import Request
+
+            async def get_arqredis(request: Request) -> ArqRedis:
+                return request.state.arqredis
+
+    .. tab-item:: Litestar
+        :sync: litestar
+
+        .. code-block:: python
+
+            from litestar import State
+
+            async def get_arqredis(state: State) -> ArqRedis:
+                return state.arqredis
+
+Post json
+~~~~~~~~~
+In FastAPI, you pass the JSON object directly as a parameter to the endpoint, you are forced to use pydantic. In Litestar, you use the `data` keyword argument and can pass whatever library supported.
+
+.. tab-set::
+
+    .. tab-item:: FastAPI
+        :sync: fastapi
+
+        .. code-block:: python
+
+
+            class ObjectType(BaseModel):
+                name: str
+
+            @app.post("/items/")
+            async def create_item(object_name: ObjectType) -> dict[str, str]:
+                return {"name": object_name.name}
+
+    .. tab-item:: Litestar
+        :sync: litestar
+
+        .. code-block:: python
+
+            from litestar import Litestar, post
+            from pydantic import BaseModel
+
+            class ObjectType(BaseModel):
+                name: str
+
+            @post("/items/")
+            async def create_item(data: ObjectType) -> dict[str, str]:
+                return {"name": data.name}
+
+
+Media type
+~~~~~~~~~~
+In FastAPI, you specify the response type using `-> HTMLResponse` for instance. In Litestar, you use the `media_type` parameter in the decorator.
```

Comment:
I think FastAPI also allows setting `media_type` on the response object?

Reply:
ok thinking about this now it doesnt make too much sense to mention that since it is incorrect on the fastapi side and it's pretty similar, so i deleted that